doc/2/guides/develop-on-kuzzle/3-event-system/index.md

```diff
@@ -0,0 +1,155 @@
+
+::: info
+The complete list of events is available here: [Internal Events List](/core/2/some-link)
+:::
+
+## Pipe
+
+<!-- Duplicate with guides/getting-started/8-customize-api-behavior -->
+
+Kuzzle allows to modify API actions behavior with a **very precise middleware-like system**.  
+
+This system allows to **modify the execution flow of requests** processed by Kuzzle.
+
+Pipes are **functions plugged to events**, called **synchronously** by Kuzzle, and **receiving information** regarding that event.
+
+Pipes can:
+  - **Change the received information**. Kuzzle will use the updated information upon resuming the task
+  - **Abort a task**. If a pipe throws an error, Kuzzle interrupts the task, and forwards a standardized version of the thrown error to the originating client
+
+![pipe workflow](./pipes-workflow.png)
+
+::: warning
+Each event carries a different payload. **This payload must be returned by the pipe function** so Kuzzle can continue its execution process.
+:::
+::: warning
+The execution order of the pipes is decided by Kuzzle at runtime. Your pipes should be independent from one another.
+:::
+Examples of pipes usage:
+ - [dynamic right restrictions](/core/2/some-link)
+ - [synchronize with another database](/core/2/some-link)
+ - [hidding sensitive information from the response](/core/2/some-link)
+
+
+### Registering a pipe
+
+<!-- Duplicate with guides/getting-started/8-customize-api-behavior -->
+
+We need to use the [Backend.pipe.register](/core/2/some-link) method to register new pipes. This method takes an event name as its first parameter, followed by the pipe handler function.
+
+Each event has a different payload.  
+The pipe handler function **must return a promise resolving to the received payload**.  
+
+::: info
+When an event has more than one payload then only the first argument of the handler function must be returned. (e.g. [Generic Document Events](/core/2/some-link))
+:::
+
+**Example:** _Changing the result of the [server:now](/core/2/api/controllers/server/now) API action_
+
+```js
+app.pipe.register('server:afterNow', async (request: Request) => {
+  request.result.now = (new Date()).toUTCString()
+
+  return request
+})
+```
+
+::: warning
+As pipes are executed synchronously by Kuzzle, they can increase the execution time of a request.  
+A pipe that takes a long time to execute will generate an alert message in the logs.
+This warning can be configured under the [plugins.pipeWarnTime](/core/2/main-concepts/configuration) configuration key.
+:::
+
+### Aborting a task
+
+When the pipe handler function returns a rejected promise, Kuzzle abort the current task.
+
+If the error is one of the [available default errors](/core/2/some-link) then the response returned to the client will contain the error as is, otherwise the error will be wrapped in a [PluginInternalError](/core/2/some-link).
```

Comment:
```suggestion
If the error is one of the [available default errors](/core/2/some-link) then the response returned to the client will contain the error as is, otherwise the error will be wrapped in a [PluginImplementationError](/core/2/some-link) error.
```